libr/anal/fcn.c

```diff
@@ -264,76 +264,55 @@ static ut64 search_reg_val(RAnal *anal, ut8 *buf, ut64 len, ut64 addr, char *reg
 #define gotoBeach(x) ret=x;goto beach;
 #define gotoBeachRet() goto beach;
 
-R_API void fill_args (RAnal *anal, RAnalFunction *fcn, RAnalOp *op) {
-	char* varname;
-	switch (op->stackop) {
-	case R_ANAL_STACK_NULL:
-	case R_ANAL_STACK_NOP:
-	case R_ANAL_STACK_ALIGN:
-	case R_ANAL_STACK_INC: {
-		char bp[50]={',',0};
-		strcat (bp, anal->reg->name[R_REG_NAME_BP]);
-		strcat (bp, ",+,[4],");
-		//TODO char, short, and long in case of 64bit it wont be [4]
-		//This will be some hints for type propagation imho
-		char *esil_buf, *ptr_end, *addr, *op_esil;
-		st64 ptr;
-		op_esil = r_strbuf_get (&op->esil);
-		if (!op_esil) {
-			return;
-		}
-		esil_buf = strdup (op_esil);
-		if (!esil_buf) {
-			return;
-		}
-		ptr_end = strstr (esil_buf, bp);
-		if (!ptr_end) {
-			free (esil_buf);
-			break;
-		}
-		*ptr_end = 0;
-		addr = ptr_end;
-		while ((*addr != '0' || *(addr+1) != 'x') &&
-			addr >= esil_buf && *addr != ',' ) {
-			addr--;
-		}
-		if (strncmp (addr, "0x", 2)) {
-			free (esil_buf);
-			break;
-		}
-		ptr = (st64)r_num_get (NULL, addr);
-		varname = get_varname (anal, ARGPREFIX, R_ABS (ptr));
-		r_anal_var_add (anal, fcn->addr, 1, ptr, fcn->call, NULL, anal->bits/8, varname);
-		r_anal_var_access (anal, fcn->addr, fcn->call, 1, ptr, 0, op->addr);
+void extract_arg (RAnal *anal, RAnalFunction *fcn, RAnalOp *op, char *reg, char *sign, char type) {
+	char *varname, *esil_buf, *ptr_end, *addr, *op_esil;
+	st64 ptr;
+	char sig[50] = {',', 0};
```

Comment:
use the heap. not sig[50]. that {',',0} looks weird too. what are you trying to do here?